lib/Dialect/LWE/IR/NewLWEAttributes.td

```diff
@@ -271,30 +271,27 @@ def LWE_PlaintextSpaceAttr : AttrDef<LWE_Dialect, "PlaintextSpace"> {
 def LWE_KeyAttr : AttrDef<LWE_Dialect, "Key"> {
   let mnemonic = "key";
   let description = [{
-    An attribute describing the key used for encrypting the ciphertext.
-
-    This attribute includes a key identifier for the original key used to
-    encrypt the secret key.
-
-    The `key_size` parameter is used to describe the number of polynomials of
-    the secret key. This is typically $1$ for RLWE ciphertexts and greater than
-    $1$ for LWE instances. A ciphertext encrypted with a `key_size` of $k$ will
-    have size $k+1$.
-
-    The key basis describes the inner product used in the phase calculation in
-    decryption. This attribute is only supported for RLWE ciphertexts whose
-    `key_size` is $1$. An RLWE ciphertext is canonically encrypted against key
-    basis `(1, s)`. After a multiplication, its size will increase and the basis
-    will be `(1, s, s^2)`. The array that represents the key basis is
-    constructed by listing the powers of `s` at each position of the array. For
-    example, `(1, s, s^2)` corresponds to `[0, 1, 2]`, while `(1, s^2)`
-    corresponds to `[0, 2]`.
+    An attribute describing the key used for decrypting the ciphertext.
+
+    The key attribute describes the key with which decryption is performed. For
+    example, if the decryption of a ciphertext $c = (c_0(x), c_1(x))$ is
+    performed by computing the inner product $(c_0(x), c_1(x)) \cdot (1, s(x))$
+    then the decryption key has a basis $(1, s(x))$.
+
+    The `rotation_basis` describes the powers of `x` at each position of the
+    array, so $(1, s(x))$ becomes $[0, 1]$. After a Galois rotation mapping $x
+    -> x^i$, the key will have a new basis $(1, s(x^i))$, which will be
+    represented by $[0, i]$.
```

Comment:
I was thinking that should we record the `x^i` or `x^5^i`, namely the automorphism index or the slot rotation index. The former would be hard to read. For OpenFHE, to map rotation index (like 1, 2, -1) to automorphism index, a `FindAutomorphismIndex` method is used (seems that they have different impl e.g. `FindAutomorphismIndex2nComplex` 
 for CKKS), which is quite some computing, and we need a inverse function for it when we analyse the rotation index in some optimization.
 
 Also, for rotation index like `-1`, the automorphism index could be large.

Reply:
Oh, that also makes me wonder what a key basis would look like for a more complex automorphism (hypercube view of a ctxt rather than array view)...

Reply:
If this works for what we need right now, I think it would be safe to merge and revisit a more complex version later.

For one, it would be reasonable to implement a custom parser/printer for the basis, so that we can have a better readability/storage tradeoff.

In particular, could we extend/re-use the polynomial parser/printer for this (in a future PR)? It already supports parsing an expression like `1+s+s^2`, but it does not currently support something like `1 + s(x^5)` or `1 + s^2(x^5)`.

Reply:
Just to resolve this - maybe the simplest thing for what we need right now is just the slot rotation index? for openfhe codegen finding the automorphism index is simple enough, but you're right we need to perform a calculation to figure out the correct slot rotation index that would correspond to the automorphism when we lower to LWE. I can attach a TODO here though and clarify that in the docs

Reply:
SGTM